modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/filename/PdsConsistentIdProcessor.java

```diff
@@ -73,6 +77,27 @@ public PdsConsistentIdProcessor(final GridKernalContext ctx) {
         return settings;
     }
 
+    /** {@inheritDoc} */
+    @Override public NodeFileTree fileTree() {
+        if (ft == null) {
```

Comment:
Looks like we can put initialization to constructor. I'm not sure, but this code might be non-thread safe, it's pretty unclear how to understand it.